doc: openthread: add power consumption data (#10030)
nrfconnect/sdk-nrf: doc/nrf/ug_thread_power_consumption.rst

.. _thread_power_consumption:

OpenThread power consumption
############################

.. contents::
   :local:
   :depth: 2

This page provides information about the amount of power used by Sleepy End Devices (SEDs) and Synchronized Sleepy End Devices (SSEDs) that use the OpenThread stack.

.. _thread_power_consumption_methodology:

Measurements methodology
************************

The measurement setup consists of:

* An nrf52840dk_nrf52840 board used as a Thread leader.
* A DUT board used as a Thread child.
* A `Power Profiler Kit II (PPK2)`_ attached to the DUT according to the instructions in its Quick start guide.

The leader board is flashed with the regular :ref:`Thread CLI sample <ot_cli_sample>` firmware.

> The DUT board is flashed with the :ref:`Thread CLI sample low power mode <ot_cli_sample_low_power>` firmware.

Reviewer: Would it make sense to enable RAM Power Down lib: https://github.com/nrfconnect/sdk-nrf/blob/main/lib/ram_pwrdn/Kconfig#L7

Author: This is already done: https://github.com/nrfconnect/sdk-nrf/blob/main/samples/openthread/cli/overlay-low_power.conf#L9, https://github.com/nrfconnect/sdk-nrf/blob/main/samples/openthread/cli/src/low_power.c#L27

I'll add the building instructions to make clear which code is being used.

         +===============================+=======+
         | Supply voltage [V]            |   3.0 |
         +-------------------------------+-------+
         | Transmission power [dBm]      |     0 |
         +-------------------------------+-------+
         | CSL period [ms]               |  1000 |
         +-------------------------------+-------+
         | CSL timeout [s]               |    20 |
         +-------------------------------+-------+
         | Parent's CSL uncertainty [μs] |  ±200 |
         +-------------------------------+-------+
         | Parent's CSL accuracy [ppm]   |    ±5 |
         +-------------------------------+-------+

      .. table:: Power consumption

         +---------------------------------+-------------------+
         | Parameter                       | nrf5340dk_nrf5340 |
         +=================================+===================+
         | Average sleep current [μA]      |              4.02 |
         +---------------------------------+-------------------+
         | Average CSL receive charge [μC] |             14.31 |
         +---------------------------------+-------------------+

>          | Total charge per minute [mC]    |              1.25 |

Reviewer: This is without any data exchange or some? In general 25% better than SED?

Author: That's true, but with some trick... I'll make more clear that the statistics are for one minute of "just listening".

The nRF5 SDK Battery life estimation white paper proposed a few scenarios with different transmission rates (30 min, 10 min...) and frame sizes, but the battery life was impacted by only about a 3%. The only big impact was for the case of 1 min rate, with a 25% decrease. So the idea is to provide just sleepy energy usage plus energy required for a transmission (to be added here later on).

The total energy used by the SSED during a minute is lower than the SED case, but only because the CSL timeout is adjusted to 20s, which is not the default. The reason is that the parent reports 5 pmm accuracy, which makes the receive window duration increase over time as long as there is no synchronization.

.. _thread_power_consumption:

OpenThread power consumption
############################

.. contents::
   :local:
   :depth: 2

> This page provides information about the amount of power used by Sleep End Devices (SEDs) and Synchronized Sleep End Devices (SSEDs) using OpenThread stack.

Reviewer 2: ```suggestion
This page provides information about the amount of power used by Sleepy End Devices (SEDs) and Synchronized Sleepy End Devices (SSEDs) that use the OpenThread stack.
```
End change is so that the sentence can't be read "provides information [...] using the OpenThread stack".

.. _thread_power_consumption_methodology:

Measurements methodology
************************

The measurement setup consists of:

> * An nrf52840dk_nrf52840 board used as Thread leader.

Reviewer 2: ```suggestion
* An nrf52840dk_nrf52840 board used as a Thread leader.
```

> * A DUT board used as Thread child.

Reviewer 2: ```suggestion
* A DUT board used as a Thread child.
```
What's DUT?

> * A `Power Profiler Kit II (PPK2)`_ attached to the DUT according to its "Quick start" guide.

Reviewer 2: ```suggestion
* A `Power Profiler Kit II (PPK2)`_ attached to the DUT according to the instructions in its Quick start guide.
```

The leader board is flashed with the regular :ref:`Thread CLI sample <ot_cli_sample>` firmware.
The DUT board is flashed with the :ref:`Thread CLI sample low power mode <ot_cli_sample_low_power>` firmware.

After the Thread network is enabled on the leader, the child is configured with the desired parameters.

> Once attached to the network the child device will switch to sleep mode (disabling all peripherals) and wake up according to the configured period only.

Reviewer 2: ```suggestion
Once attached to the network, the child device will switch to sleep mode (disabling all peripherals) and wake up according to the configured period only.
```
Would "schedule" work here instead of "period"?
Alternatively, it could just "...according to its configuration:" and have the next two sentences as bullet points.

For the SED case, it will regularly send MAC Data Poll Commands.
For the SSED case, it will regularly switch to radio receive mode.
The PPK2 is used to collect power consumption data during at least 60 seconds.
Afterwards, the CSV data is parsed with a script to produce the below results.

.. _thread_power_consumption_data:

Power consumption data
**********************

> 

Reviewer 2: I'd add the following introduction: "The following tables show the power consumption measured with the configuration used." Or something else along those lines.

> .. tabs::

Reviewer 2: I think this would work as only one tabs contruct with a "Thread 1.3 SED" tab and a "Thread 1.3 SSED" tab, each containing the respective configuration table and power consumption table.

Author: Yes, it's as intended: https://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-10030/nrf/ug_thread_power_consumption.html#power-consumption-data

I've removed the tables' descriptions since they are repetitive with the tab's title.

Reviewer 2: It does have some unintended behavior with being able to change the lower set of tabs (Power consumption) without the top ones changing. Using just the top one seems to change both sets.

.. _thread_power_consumption:

OpenThread power consumption
############################

.. contents::
   :local:
   :depth: 2

This page provides information about the amount of power used by Sleepy End Devices (SEDs) and Synchronized Sleepy End Devices (SSEDs) that use the OpenThread stack.

.. _thread_power_consumption_methodology:

Measurements methodology
************************

The measurement setup consists of:

* An nrf52840dk_nrf52840 board used as a Thread leader.
* A DUT board used as a Thread child.
* A `Power Profiler Kit II (PPK2)`_ attached to the DUT according to the instructions in its Quick start guide.

The leader board is flashed with the regular :ref:`Thread CLI sample <ot_cli_sample>` firmware.

     .. code-block::

        cd ncs/nrf/samples/openthread/cli/
        west build -b nrf52840dk_nrf52840 -- -DOVERLAY_CONFIG="overlay-ci.conf;overlay-logging.conf"

The DUT board is flashed with the :ref:`Thread CLI sample low power mode <ot_cli_sample_low_power>` firmware.

> In the building command below _build-target_ is replaced with the build target name of the DUT.

Reviewer 2: ```suggestion
In the build command below, replace ``build-target`` with the build target name of the DUT.
```

>      .. code-block::
> 
>         cd ncs/nrf/samples/openthread/cli/
>         west build -b build-target -- -DOVERLAY_CONFIG="overlay-ci.conf;overlay-low_power.conf" -DDTC_OVERLAY_FILE="low_power.overlay"

Reviewer 2: ```suggestion
   .. code-block::

      cd ncs/nrf/samples/openthread/cli/
      west build -b build-target -- -DOVERLAY_CONFIG="overlay-ci.conf;overlay-low_power.conf" -DDTC_OVERLAY_FILE="low_power.overlay"
```
Don't need quite that much indentation.

After the Thread network is enabled on the leader, the child is configured with the desired parameters.

> Once attached to the network, the child device will switch to sleep mode (disabling all peripherals) and wake up according to its configuration:

Reviewer: ```suggestion
Once attached to the network, the child device will switch to sleep mode (disabling all peripherals except RTC) and wake up according to its configuration:
```

* For the SED case, it will regularly send MAC Data Poll Commands.
* For the SSED case, it will regularly switch to radio receive mode.

The PPK2 is used to collect power consumption data during at least 60 seconds.

> During this time no frames are being sent from the leader to the child.

Reviewer: ```suggestion
During this time no data frames are being sent from the leader to the child.
```
